frontend/src/apiMocks/handlers.ts

```diff
@@ -0,0 +1,237 @@
+import { rest, RestHandler, RestRequest } from "msw";
+import { CustomAliasData, RandomAliasData } from "../hooks/api/aliases";
+import { ProfileData } from "../hooks/api/profile";
+import { RuntimeData } from "../hooks/api/runtimeData";
+import { mockIds, domainaddresses, profiles, relayaddresses } from "./mockData";
+
+export function getHandlers(
+  defaultMockId: null | typeof mockIds[number] = null
+): RestHandler[] {
+  const handlers: RestHandler[] = [];
+
+  const getMockId = (req: RestRequest): typeof mockIds[number] | null => {
+    const authHeader = req.headers.get("Authorization");
+    if (typeof authHeader !== "string") {
+      return defaultMockId;
+    }
+    const token = authHeader.split(" ")[1];
+    return mockIds.find((mockId) => mockId === token) ?? defaultMockId;
+  };
+
+  const addGetHandler: (...args: Parameters<typeof rest.get>) => void = (
+    path,
+    resolver
+  ) => {
+    handlers.push(rest.get(path, resolver));
+    handlers.push(rest.get(`http://127.0.0.1:8000${path}`, resolver));
+  };
+  const addPatchHandler: (...args: Parameters<typeof rest.patch>) => void = (
+    path,
+    resolver
+  ) => {
+    handlers.push(rest.patch(path, resolver));
+    handlers.push(rest.patch(`http://127.0.0.1:8000${path}`, resolver));
+  };
+  const addPostHandler: (...args: Parameters<typeof rest.post>) => void = (
+    path,
+    resolver
+  ) => {
+    handlers.push(rest.post(path, resolver));
+    handlers.push(rest.post(`http://127.0.0.1:8000${path}`, resolver));
+  };
+  const addDeleteHandler: (...args: Parameters<typeof rest.delete>) => void = (
+    path,
+    resolver
+  ) => {
+    handlers.push(rest.delete(path, resolver));
+    handlers.push(rest.delete(`http://127.0.0.1:8000${path}`, resolver));
+  };
+
+  addGetHandler("/accounts/logout", (_req, res, ctx) => {
+    return res();
+  });
+  addGetHandler("/api/v1/runtime_data", (_req, res, ctx) => {
+    const runtimeData: RuntimeData = {
+      FXA_ORIGIN: "https://example.com",
+      GOOGLE_ANALYTICS_ID: "UA-123456789-0",
+      PREMIUM_PRODUCT_ID: "prod_123456789",
```

Comment:
Question: Can we move these IDs to env variables? Not that I expect stuff to change often, but its' a risk putting the static values here.

Reply:
We can, but if you look at the actual IDs, these are fake values, and their actual value doesn't matter. So should be too risky, I think.